chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -988,6 +1017,13 @@ impl<'a, S: BlockchainStorageWrite, V: TransactionVerificationStrategy> Chainsta
         new_tip_block_index: &BlockIndex,
         new_tip: &WithId<Block>,
     ) -> Result<(), BlockError> {
+        debug_assert!(
+            new_tip_block_index.status().is_ok()
+                && new_tip_block_index.status().last_valid_stage()
+                    >= BlockValidationStage::CheckBlockOk
+        );
+        // FIXME: the block should already be at the CheckBlockOk stage at this moment,
```

Comment:
I don't mind removing this. But please add a check for the state instead (even if it's an assert, which is relatively dangerous, but OK for testnet), as a guarantee that it's what we expect it's.